### ISSUE TITLE
Changes to DAG in the Production UI requires approval

### ISSUE THREAD

**Issue by afsalabdup** [apache/airflow#9302]
Any changes that occur in UI to DAGs fall under audit requirement. So we need an approval from a manager or designated approver for all manual actions like rerun, trigger a dag etc in the UI. 

All manual action buttons in the UI should be disabled by default. We may have to include a new button in UI, so that when we click on that new button, it will go and check any open ServiceNow requests/tickets, if there is any then only user will be able to do any manual action in the UI. Once the action is done then all buttons should be disabled again.

**Comment by tooptoop4** [comment]
since when can u edit dag in the UI? @afsalabdup

**Comment by afsalabdup** [comment]
@tooptoop4 - I meant manual actions to tasks in DAG like rerun, mark success etc

**Comment by tooptoop4** [comment]
use rbac ui, can make it readonly for some ppl

**Comment by afsalabdup** [comment]
Even if you have write access, you need approval before you do manual actions in UI, that's my question.

**Comment by ashb** [comment]
This is out of scope for OSS Airflow. If you wanted such a feature you could probably build it via a plugin.
